File: day-02/part-01/go/02-01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	data, err := ioutil.ReadFile("../input.txt")
	if err != nil {
		fmt.Println(err)
	}
	slicedData := strings.Split(string(data), "\n")
	directions := make([]string, len(slicedData))
	distances := make([]int, len(slicedData))
	for i, v := range slicedData {
		line := strings.Fields(v)
		directions[i] = line[0]
		distances[i], _ = strconv.Atoi(line[1])
	}

	vertical := 0
	horizontal := 0

	for i := 0; i < len(slicedData); i++ {
		if directions[i] == "forward" {
			horizontal += distances[i]
		} else if directions[i] == "up" {
			vertical -= distances[i]
		} else if directions[i] == "down" {
			vertical += distances[i]
		}
	}

	fmt.Println(vertical * horizontal)
}

File: day-02/part-01/rust/02-01.rs
use std::fs;

fn main() {
  let data = fs::read_to_string("../input.txt").expect("Something went wrong reading the file");
  // split each line into a tuple of a string and an integer
  let lines: Vec<(String, i32)> = data.lines().map(|line| {
    let mut parts = line.split_whitespace();
    (parts.next().unwrap().to_string(), parts.next().unwrap().parse::<i32>().unwrap())
  }).collect();

  let mut vertical = 0;
  let mut horizontal = 0;

  for (direction, distance) in lines {
    if direction == "forward" {
      vertical += distance;
    } else if direction == "down" {
      horizontal += distance;
    } else if direction == "up" {
      horizontal -= distance;
    }
  }

  println!("{}", vertical*horizontal);
}
